src/owr/ObolErc1155Recipient.sol

```diff
@@ -0,0 +1,288 @@
+    lastId = 1;
+    _baseUri = baseUri_;
+    depositContract = IDepositContract(_depositContract);
+
+    _initializeOwner(_owner);
+  }
+
+  receive() external payable {}
+
+  /// -----------------------------------------------------------------------
+  /// functions - view & pure
+  /// -----------------------------------------------------------------------
+  /// @dev Returns the Uniform Resource Identifier (URI) for token `id`.
+  function uri(uint256 id) public view override returns (string memory) {
+    return string(abi.encodePacked(_baseUri, LibString.toString(id)));
+  }
+
+  /// @dev Returns true if `msg.sender` is the owner of the contract
+  function isOwnerOf(uint256 _tokenId) public view returns (bool) {
+    return balanceOf(msg.sender, _tokenId) > 0;
+  }
+
+  /// @dev Returns max supply for id
+  function getMaxSupply(uint256 _tokenId) public view returns (uint256) {
+    uint256 _partition = tokensPartition[_tokenId];
+    return partitions[_partition].maxSupply;
+  }
+
+  /// -----------------------------------------------------------------------
+  /// functions - public & external
+  /// -----------------------------------------------------------------------
+
+  /// @notice creates a new partition
+  /// @dev TODO: TBD DOS risk
+  /// @param maxSupply the maximum number of unique tokens
+  /// @param owr the Optimistic Withdrawal Recipient address
+  function createPartition(uint256 maxSupply, address owr) external {
+    uint256 _id = partitionCount;
+    partitions[_id] = Partition({maxSupply: maxSupply, owr: owr, operator: msg.sender});
+    owrsPartition[owr] = _id;
+
+    partitionCount++;
+    emit PartitionCreated(owr, _id, maxSupply);
+  }
+
+  /// @notice mints a new token and deposits to ETH deposit contract
+  /// @param _partitionId the partition to assign it to
+  /// @param depositInfo deposit data needed for `DepositContract`
+  /// @return mintedId id of the minted NFT
+  function mint(uint256 _partitionId, DepositInfo calldata depositInfo) external payable returns (uint256 mintedId) {
+    // validation
+    if (partitions[_partitionId].owr == address(0)) revert PartitionNotValid();
+    if (partitionTokens[_partitionId].length + 1 > partitions[_partitionId].maxSupply) revert PartitionSupplyReached();
+    if (msg.value != ETH_DEPOSIT_AMOUNT) revert DepositAmountNotValid();
+
+    // retrieve id
+    mintedId = _incrementId();
+
+    // add partition details
+    partitionTokens[_partitionId].push(mintedId);
+    tokensPartition[mintedId] = _partitionId;
+
+    // increase total supply
+    totalSupply[mintedId]++;
+    totalSupplyAll++;
+
+    // deposit to ETH deposit contract
+    depositContract.deposit{value: ETH_DEPOSIT_AMOUNT}(
+      depositInfo.pubkey, depositInfo.withdrawal_credentials, depositInfo.sig, depositInfo.root
+    );
+
+    // mint to sender
+    _mint(msg.sender, mintedId, 1, "");
+    ownerOf[mintedId] = msg.sender;
```

Comment:
vulnerable to re-entrant attack. we should follow CEI pattern. Also, we should verify that  the pubkey has not already been minted

Reply:
What do you mean by verifying the pub key?